### ISSUE TITLE
Script that gives an error in R-Instat

### ISSUE THREAD

**Issue by rdstern** [IDEMSInternational/R-Instat#8942]
@lloyddewit it is from the gt package example and is the one on [clinical trials.](https://gt.rstudio.com/articles/case-study-clinical-tables.html)

Here is the script:
```
custom_summary <- function(df, group_var, sum_var) {
  
  group_var <- rlang::ensym(group_var)
  sum_var <- rlang::ensym(sum_var)
  
  is_categorical <- 
    is.character(eval(expr(`$`(df, !!sum_var)))) |
    is.factor(eval(expr(`$`(df, !!sum_var)))) 
  
  if (is_categorical) {

    category_lbl <- 
      sprintf("%s, n (%%)", attr(eval(expr(`$`(df, !!sum_var))), "label"))

    df_out <-
      df |>
      dplyr::group_by(!!group_var)  |> 
      dplyr::mutate(N = dplyr::n()) |> 
      dplyr::ungroup() |> 
      dplyr::group_by(!!group_var, !!sum_var) |> 
      dplyr::summarize(
        val = dplyr::n(),
        pct = dplyr::n()/mean(N),
        .groups = "drop"
      ) |> 
      tidyr::pivot_wider(
        id_cols = !!sum_var, names_from = !!group_var,
        values_from = c(val, pct)
      ) |> 
      dplyr::rename(label = !!sum_var) |> 
      dplyr::mutate(
        across(where(is.numeric), ~ifelse(is.na(.), 0, .)),
        category = category_lbl
      )

  } else {

    category_lbl <-
      sprintf(
        "%s (%s)",
        attr(eval(expr(`$`(df, !!sum_var))), "label"),
        attr(eval(expr(`$`(df, !!sum_var))), "units")
      )

    df_out <- 
      df |> 
      dplyr::group_by(!!group_var) |> 
      dplyr::summarize(
        n = sum(!is.na(!!sum_var)),
        mean = mean(!!sum_var, na.rm = TRUE),
        sd = sd(!!sum_var, na.rm = TRUE),
        median = median(!!sum_var, na.rm = TRUE),
        min = min(!!sum_var, na.rm = TRUE),
        max = max(!!sum_var, na.rm = TRUE),
        min_max = NA,
        .groups = "drop"
      ) |> 
      tidyr::pivot_longer(
        cols = c(n, mean, median, min_max),
        names_to = "label",
        values_to = "val"
      ) |> 
      dplyr::mutate(
        sd = ifelse(label == "mean", sd, NA),
        max = ifelse(label == "min_max", max, NA),
        min = ifelse(label == "min_max", min, NA),
        label = dplyr::recode(
          label,
          "mean" = "Mean (SD)",
          "min_max" = "Min - Max",
          "median" = "Median"
        )
      ) |> 
      tidyr::pivot_wider(
        id_cols = label,
        names_from = !!group_var,
        values_from = c(val, sd, min, max)
      ) |> 
      dplyr::mutate(category = category_lbl)
  }

  return(df_out)
}

adsl_summary <- 
  dplyr::filter(rx_adsl, ITTFL == "Y") |> 
  (\(data) purrr::map_df(
    .x = dplyr::vars(AGE, AAGEGR1, SEX, ETHNIC, BLBMI),
    .f = \(x) custom_summary(df = data, group_var = TRTA, sum_var = !!x)
  ))()
```

The first part, which presumably produces the custom_summary function seems to run ok.
The second part uses the custom_summary function and gives an error.   I get the same strange error if I just type custom_summary.

I am not quite sure what it is doing here?  It is @lilyclements sort of R, rather than mine!

**Comment by lloyddewit** [comment]
@rdstern 
Thank you for the scripts.
If anyone finds a script that fails in R-Instat, then it's best to first test the script in RStudio. If it fails in RStudio then it's likely a script issue rather than an R-Instat issue.
If it runs in RStudio but not in R-Instat, then it would be helpful to provide a screenshot of the R-Instat error(s).
I tried to run the scripts in RStudio and they both failed. If you or @lilyclements can get these scripts to work in RStudio, and they still fail in R-Instat, then I can investigate further.
Thanks

**Comment by lloyddewit** [comment]
@rdstern
Thank you for the scripts.
If anyone finds a script that fails in R-Instat, then it's usually best to first test the script in RStudio. If it fails in RStudio then it's likely a script issue rather than an R-Instat issue.
If it runs in RStudio but not in R-Instat, then it would be helpful to provide a screenshot of the R-Instat error(s).
I tried to run the scripts in RStudio and they both failed. Normally, I would ask if you or @lilyclements could get these scripts to work in RStudio, and if they still failed in R-Instat, then I would investigate further.

However, I was curious and tried the scripts in R-Instat. Surprisingly, they ran better in R-Instat than RStudio. I assume that my R-Instat R environment must be slightly different to my RStudio R environment.

**Script 1**

I got the following error:

![image](https://github.com/IDEMSInternational/R-Instat/assets/57253949/87bc7648-a2dc-4b97-a257-1b0d6852a21e)

R-Instat is correct to report this error because the `md` function currently in scope requires a `y` parameter:

![image](https://github.com/IDEMSInternational/R-Instat/assets/57253949/91c792ff-b803-43b1-9e52-7e9d2b9a2172)

If we remove the `md` function then the whole of script 1 runs correctly.

![image](https://github.com/IDEMSInternational/R-Instat/assets/57253949/38bc973a-868c-4b34-865f-7d4dc9ecc4b6)

So in summary, I don't think there's anything I need to fix in R-Instat for script 1.

**Script 2**

As you said, R-Instat defines the function correctly.
However, it cannot parse the function call. Specifically it cannot parse the highlighted text below.
@lilyclements Do you know what the highlighted parts do and specifically what `\` does? Is this valid R that we need to be able to parse in R-Instat?

![image](https://github.com/IDEMSInternational/R-Instat/assets/57253949/6988c2e4-a0a3-4f90-98c0-9bbb79297334)

Thanks for your help

**Comment by lloyddewit** [comment]
@rdstern 
For the second script, was there also a potential issue copying/pasting from the website?
For example `!` is changed to `!!` and the pipe symbol looks strange:

![image](https://github.com/IDEMSInternational/R-Instat/assets/57253949/4d1bc9e1-7c2b-4c51-a451-591a01e38701)

**Comment by rdstern** [comment]
@lloyddewit thank you very much for the detective work.  The md was the big one, because it happened regularly.  This is now solved.  And I should have guessed!
a) Another clue that there was going to be an obvious solution was that I couldn't find evidence on the web that others had trouble with this function.
b) The answer is that R-Instat (and our version of RStudio) must have another package that also has an `md` command and that we load before the gt package.  I can't find which package it is.  But - in hindsight of course - that is an obvious possibility.

One solution is to put gt::md(  ), each time.  But running `md <- gt::md` once is also sufficient.  Then we can leave the resulting code as it is.  That's ok (once you know)  and is like having to include the `library(gt) `command in a script.
